Updated code with langchain, openai, pinecone packages

ddl-aambekar opened the pull request.

ddl-aambekar committed:
Updated code with langchain, openai, pinecone packages

ddl-subir-m reviewed (commented):
@ddl-aambekar the environment.json and the project.json files in the template_defintion folder need to be updated as well (details in runbook [here](https://dominodatalab.atlassian.net/wiki/spaces/ENG/pages/2557313178/AI+Hub+Runbook) ). The commit id in the project.json needs to be updated  (we can do it after merging the PR too). While you are it, do you mind changing the environment.json and adding a dockerfiIeLayer.txt in the template_definiton directory?I  noticed this project is using a GPU env but we can use a non GPU DSE and add the requirements from the readme to the dockerfileLayer.txt. I also think we dont need streamlit-chat anymore since the latest version of streamlit now has chat components. If you remove streamlit-chat also remove it from the basic_checks.py in the tests folder

ddl-aambekar committed:
Updated files per review comments

ddl-aambekar committed:
Removed streamlit-chat test from basic_checks.py

ddl-aambekar commented:
Thank you for the review comments, @ddl-subir-m. I've modified the environment.json to use the DSE instead of the GPU DSE, and added requirements from readme to the dockerfileLayer.txt. I also removed streamlit-chat from requirements.txt and basic_checks.py. I tested the notebooks and the app and they worked with the DSE + dockerfileLayer changes as expected.

ddl-subir-m reviewed (commented):
nitpick: its more efficient to install all the packages in one RUN pip install instead of creating a new layer with every RUN command

ddl-aambekar committed:
Update dockerfiIeLayer.txt

ddl-aambekar committed:
Update README.md
